src/cljs/rems/administration/create_workflow.cljs

```diff
@@ -295,43 +294,38 @@
    [workflow-type-description (text :t.create-workflow/auto-approve-workflow-description)]])
 
 (defn create-workflow-page []
-  (let [state (r/atom nil)
-        on-pending #(reset! state {:status :pending})
-        on-success #(reset! state {:status :saved})
-        on-error #(reset! state {:status :failed :error %})
-        on-modal-close #(do (reset! state nil)
-                            (dispatch! "#/administration/workflows"))]
+  (let [{:keys [on-pending on-success on-error state-atom] :as modal-opts}
+        (status-modal/status-modal-opts
+         {:on-close-after-success #(dispatch! "#/administration/workflows")
+          :description (text :t.administration/create-workflow)})]
     (fn []
-     (let [form @(rf/subscribe [::form])
-           workflow-type (:type form)
-           loading? (rf/subscribe [::loading?])]
-       [:div
-        [administration-navigator-container]
-        [:h2 (text :t.administration/create-workflow)]
-        [collapsible/component
-         {:id "create-workflow"
-          :title (text :t.administration/create-workflow)
-          :always [:div
-                   (if @loading?
-                     [:div#workflow-loader [spinner/big]]
-                     [:div#workflow-editor
-                      (when (:status @state)
-                        [status-modal (assoc @state
-                                             :description (text :t.administration/create-workflow)
-                                             :on-close on-modal-close)])
-                      [workflow-organization-field]
-                      [workflow-title-field]
-                      [workflow-type-field]
-
-                      (case workflow-type
-                        :auto-approve [auto-approve-workflow-form]
-                        :dynamic [dynamic-workflow-form]
-                        :rounds [round-workflow-form])
-
-                      [:div.col.commands
-                       [cancel-button]
-                       [save-workflow-button #(rf/dispatch [::create-workflow
-                                                            {:request %
-                                                             :on-pending on-pending
-                                                             :on-success on-success
-                                                             :on-error on-error}])]]])]}]]))))
+      (let [form @(rf/subscribe [::form])
```

Comment:
There should be no need for inner let and outer let in this function. It's part of what creates confusion.

Reply:
Now uses only one of inner/outer let